src/float.rkt

```diff
@@ -71,8 +50,111 @@
 (define (bit-difference x y)
   (ulps->bits (+ 1 (abs (ulp-difference x y)))))
 
-(define (sample-float)
-  (floating-point-bytes->real (integer->integer-bytes (random-exp 32) 4 #f)))
+(define (ordinary-value? x)
+  (match x
+    [(? real?)
+     (not (or (infinite? x) (nan? x)))]
+    [(? complex?)
+     (and (ordinary-value? (real-part x)) (ordinary-value? (imag-part x)))]
+    [(? boolean?)
+     true]
+    [(? posit8?)
+     (not (set-member? (representation-special-values posit8) x))]
+    [(? posit16?)
+     (not (set-member? (representation-special-values posit16) x))]
+    [(? posit32?)
+     (not (set-member? (representation-special-values posit32) x))]
+    [_ true]))
+
+(module+ test
+  (check-true (ordinary-value? 2.5))
+  (check-false (ordinary-value? +nan.0))
+  (check-false (ordinary-value? -inf.f)))
+
+(define (=-or-nan? x1 x2)
+  (cond
```

Comment:
I'm surprised this doesn't work on boolean values. Is that a mistake, an oversight, or is that on purpose?

Reply:
This is another weird case where we mixed up types in the past. I think it makes the most sense to have `=-or-nan?` only operate on number types (which is currently the only way we call it), although it doesn't really hurt to have it operate on bools in this case as well. There are a few functions like this that got moved over to `float.rkt` where it's not clear to me if the right call is to generalize the function to fit all the types herbie supports, or to leave this as something that operates exclusively on numbers (and maybe to even separate them out into real and complex versions once we hit that phase in the interface changes.)